com.unity.multiplayer.mlapi/Prototyping/NetworkNavMeshAgent.cs

```diff
@@ -47,9 +47,50 @@ private void Awake()
             m_Agent = GetComponent<NavMeshAgent>();
         }
 
+        private ClientRpcParams m_ClientParams;
         private Vector3 m_LastDestination = Vector3.zero;
         private float m_LastCorrectionTime = 0f;
 
+        public override void NetworkStart()
+        {
+            base.NetworkStart();
+
+            if (IsOwner)
+            {
+                CacheClientParams();
+
+                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnected;
```

Comment:
There is some issues with how these events are subscribed/unsubscribed to. NetworkStart can get called multiple times over the lifetime of an object. So spawning/unspawning an object in a row will subscribe multiple times